### ISSUE TITLE
read does not work with pathlib.Path objects

### ISSUE THREAD

**Issue by kinverarity1** [kinverarity1/lasio#277]
```
Traceback (most recent call last):
  File "src\data\make_2019_geophys_log_profiles.py", line 111, in <module>
    main()
  File "c:\devapps\python\anaconda3-32\envs\gwdev2\lib\site-packages\click\core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "c:\devapps\python\anaconda3-32\envs\gwdev2\lib\site-packages\click\core.py", line 717, in main
    rv = self.invoke(ctx)
  File "c:\devapps\python\anaconda3-32\envs\gwdev2\lib\site-packages\click\core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\devapps\python\anaconda3-32\envs\gwdev2\lib\site-packages\click\core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "src\data\make_2019_geophys_log_profiles.py", line 53, in main
    las = lasio.read(fn)
  File "c:\devapps\projects\lasio\lasio\__init__.py", line 41, in read
    return LASFile(file_ref, **kwargs)
  File "c:\devapps\projects\lasio\lasio\las.py", line 78, in __init__
    self.read(file_ref, **read_kwargs)
  File "c:\devapps\projects\lasio\lasio\las.py", line 113, in read
    file_obj, regexp_subs, value_null_subs, ignore_data=ignore_data, )
  File "c:\devapps\projects\lasio\lasio\reader.py", line 265, in read_file_contents
    for i, line in enumerate(file_obj):
TypeError: 'WindowsPath' object is not iterable
```

**Comment by kinverarity1** [comment]
Easiest way to fix would be to check with isinstance first and if it is, convert to string.

Just have to be careful with imports, that we continue to support Python 2.6 (?) and 2.7.

**Comment by kinverarity1** [comment]
Fixed by @dcslagel in #294 🎉
